File: IE 211 Fall21/src/Lab2/Sort.java
package Lab2;
import java.time.Clock; 

public class Sort {
	
	 public long runTime;
	
	 public int totalIter = 0;
	
	public Sort() {}
	
	public void InsertionSort(int[] nValues)
	{		
		Clock clock = Clock.systemDefaultZone();
		runTime  = clock.millis();
		
		for(int j = 1; j < nValues.length; j++)
		{
			int key = nValues[j];
			
			int i = j - 1;
			
			while(i >= 0 && nValues[i] > key)
			{
				nValues[i + 1] = nValues[i];
				i--;
				totalIter++;
			}
			
			nValues[i + 1] = key; 
		}
		
		runTime = clock.millis() - runTime;
		
		System.out.println();
		System.out.println("The sorted array is: ");
		
		for(int i = 0; i < nValues.length; i++)
			System.out.print(nValues[i] + " ");
		
		System.out.println();
		
		System.out.println("Total iterations: " + totalIter + " The runtime: " + runTime);
			
	}

}

File: IE211 Fall22/src/Lab2/Sort.py
import time

class Sort:
    runTime = 0
    totalIter = 0

    def ImplementAlgorithm(self, nValues):

        # Start clock
        self.runTime = time.time_ns() // 1000000

        for j in range(1, len(nValues)):
            key = nValues[j]
            i = j - 1

            while i >= 0 and nValues[i] > key:
                nValues[i + 1] = nValues[i]
                i -= 1
                self.totalIter += 1

            nValues[i + 1] = key

        # Stop clock
        self.runTime = time.time_ns() // 1000000 - self.runTime

        print(f'The sorted array is')
        print(*nValues)

        print(f'Total iterations: {self.totalIter}')
        print(f'The runtime is {self.runTime} milliseconds')